src/cmap/auth/mongo_credentials.ts

```diff
@@ -30,6 +31,7 @@ function getDefaultAuthMechanism(hello?: Document): AuthMechanism {
   return AuthMechanism.MONGODB_CR;
 }
 
+const ALLOWED_PROVIDER_NAMES = ['aws', 'azure'];
```

Comment:
```suggestion
const ALLOWED_PROVIDER_NAMES: AuthMechanismProperties['PROVIDER_NAME'][] = ['aws', 'azure'];
```

(tiny suggestion for type safety)

Reply:
Added suggestion.